File: 0009_palindrome_number/palindrome_number.js
/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function(x) {
    let original_array = String(x).split('')
    for (let i=0; i<original_array.length; i++){
        if (original_array[i] !== original_array[original_array.length - i - 1]){
            return false
        }
    }
    return true
};

console.log(isPalindrome(121))
console.log(isPalindrome(123456))
console.log(isPalindrome(1234321))
console.log(isPalindrome(-121))

File: 0009_palindrome_number/palindrome_number.go
package main

import (
    "fmt"
    "strings"
    "strconv"
)

func isPalindrome(x int) bool {
    original_array := strings.Split(strconv.Itoa(x), "")
    for i := 0; i < len(original_array); i++ {
        if original_array[i] != original_array[len(original_array) - i - 1] {
            return false
        }
    }
    return true
}

func main() {
	num := -121
	var result = isPalindrome(num)
	fmt.Println(result)
}
